src/python/pants/engine/legacy/graph.py

```diff
@@ -186,17 +187,17 @@ def inject_addresses_closure(self, addresses):
     addresses = set(addresses) - set(self._target_by_address.keys())
     if not addresses:
       return
-    dependencies = tuple(SingleAddress(a.spec_path, a.target_name) for a in addresses)
-    for _ in self._inject_specs(Specs(dependencies=tuple(dependencies))):
+    dependencies = (SingleAddress(directory=a.spec_path, name=a.target_name) for a in addresses)
+    for _ in self._inject_specs(Specs(dependencies)):
```

Comment:
I converted this into a generator. There was no need to cast to a tuple as it will be done by the `Specs` constructor. Also, this was calling `tuple()` twice.